src/rp2_common/hardware_sync/sync.c

```diff
@@ -23,8 +23,7 @@ void spin_locks_reset(void) {
 
 spin_lock_t *spin_lock_init(uint lock_num) {
     assert(lock_num < NUM_SPIN_LOCKS);
-    spin_lock_t *lock = spin_lock_instance(lock_num);
-    spin_unlock_unsafe(lock);
+    spin_unlock_unsafe(spin_lock_instance(lock_num));
```

Comment:
presumably that doesn't even compile!

Reply:
I just copied the line from `spin_lock_unclaim(uint)` for consistency reasons

Reply:
but another question, the file local variable static uint32_t claimed is uninitialized. Is that intended? Does pico automatically set 0?

Reply:
All static variable are by default initialised to zero (according to the C standard)

Reply:
right, it's static